src/common/cuda/rtc/util-inl.h

```diff
@@ -0,0 +1,380 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef MXNET_COMMON_CUDA_RTC_UTIL_INL_H_
+#define MXNET_COMMON_CUDA_RTC_UTIL_INL_H_
+
+#include <mxnet/base.h>
+
+#if MXNET_USE_CUDA
+
+namespace mxnet {
+namespace common {
+namespace cuda {
+namespace rtc {
+
+const char type_support_string[] = R"code(
+using float32 = float;
+using float64 = double;
+using float16 = half;
+using uint8 = unsigned char;
+using int8 = char;
+using int32 = int;
+using int64 = long long;
```

Comment:
will these ever be platform dependent?

Reply:
Not sure what you mean by that - are you asking about `long long` in particular or all of those? The problem with RTC is that it is not really easy to include anything external, especially system headers, so I need to use built-in types - otherwise I would use stuff like `int64_t` here.

Reply:
I see. How is the type size determined?

Reply:
They come from the standard builtin types on a given platform. That said, I see your point - I will add some static asserts to ensure that the size is what we expect.